haproxy/reload_agent.go

```diff
@@ -244,14 +241,20 @@ func (rc *reloadCache) Init(retention int) {
 	rc.lastSuccess = nil
 	rc.index = 0
 	rc.retention = retention
-	rc.channel = make(chan string)
 }
 
-func (rc *reloadCache) newReload() {
+func (rc *reloadCache) newReload() string {
```

Comment:
Upon a new reload, according to these changes, we're performing a double mutex: the first one to generate the ID, the second one for setting the next planned to reload.

We could optimize this, putting all of this logic in a single function, YMMV.

```go
func (rc *reloadCache) newReload() string {
	rc.mu.Lock()
  	defer rc.mu.Unlock()
	id := fmt.Sprintf("%s-%v", time.Now().Format("2006-01-02"), rc.index)
	rc.index++
	rc.next = id
	return rc.next
}
```

In the end, `generateID` function is just used in a single place.